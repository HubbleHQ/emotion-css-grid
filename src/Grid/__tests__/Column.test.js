/** @jsx jsx */
import { shallow, mount } from 'enzyme';

import { css, jsx } from '@emotion/core';
import { Column, GridConfigProvider } from '..';

const mockContent = <p>Hello World</p>;

beforeEach(() => {
  jest.spyOn(console, 'error')
  global.console.error.mockImplementation(() => {});
});

afterEach(() => {
  global.console.error.mockRestore();
});

describe('Grid : Column', () => {
  it('should render as expected in its default state', () => {
    const output = shallow(
      <GridConfigProvider gutterWidth={15} columnNumber={16} mobileBreakpoint={600}>
        <Column>
          {mockContent}
        </Column>
      </GridConfigProvider>,
    );
    const children = output.props().children;
    expect(children).toMatchSnapshot();
    expect(output.html()).toMatchSnapshot();
  });
  it('should render as expected when colspan is provided', () => {
    const output = shallow(
      <GridConfigProvider gutterWidth={15} columnNumber={16} mobileBreakpoint={600}>
        <Column colspan={6}>
          {mockContent}
        </Column>
      </GridConfigProvider>,
    );
    const children = output.props().children;
    expect(children).toMatchSnapshot();
    expect(output.html()).toMatchSnapshot();
  });
  it('should render as expected when provided a colspan plus styling', () => {
    const output = shallow(
      <GridConfigProvider gutterWidth={15} columnNumber={16} mobileBreakpoint={600}>
        <Column colspan={6} css={css`border: 1px solid blue;`}>
          {mockContent}
        </Column>
      </GridConfigProvider>,
    );
    const children = output.props().children;
    expect(children).toMatchSnapshot();
    expect(output.html()).toMatchSnapshot();
  });
  it('should throw an error if colspan value is greater than total number of columns', () => {
    const testFunction = () => {
      const output = mount(
        <GridConfigProvider gutterWidth={15} columnNumber={16} mobileBreakpoint={600}>
          <Column colspan={32}>
            {mockContent}
          </Column>
        </GridConfigProvider>,
      );
    };
    expect(testFunction).toThrowErrorMatchingSnapshot();
  });
});