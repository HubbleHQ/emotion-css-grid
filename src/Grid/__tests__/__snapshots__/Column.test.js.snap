// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid : Column should render as expected in its default state 1`] = `
<p>
  Hello World
</p>
`;

exports[`Grid : Column should render as expected in its default state 2`] = `
Object {
  "__EMOTION_LABEL_PLEASE_DO_NOT_USE__": "Column",
  "__EMOTION_TYPE_PLEASE_DO_NOT_USE__": "div",
  "children": <p>
    Hello World
  </p>,
  "className": "",
  "css": Object {
    "map": "/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkNvbHVtbi5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFhNkIiLCJmaWxlIjoiQ29sdW1uLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgUHJvcFR5cGVzIGZyb20gJ3Byb3AtdHlwZXMnO1xuaW1wb3J0IHsgY3NzLCBqc3ggfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuaW1wb3J0IGNvbmZpZyBmcm9tICcuL0dyaWRDb25maWcnO1xuaW1wb3J0IGNvbHVtbldpZHRoIGZyb20gJy4vY29sdW1uV2lkdGgnO1xuaW1wb3J0IHJlc3BvbnNpdmVXaWR0aCBmcm9tICcuL3Jlc3BvbnNpdmVXaWR0aCc7XG5cbmNvbnN0IHsgZ3V0dGVyV2lkdGgsIG1vYmlsZUJyZWFrcG9pbnQgfSA9IGNvbmZpZztcblxuLypcbiAqIFdlIG9taXQgZ3V0dGVycyBvbiBjb2x1bW5zIGJ5IGRlZmF1bHQgb24gbW9iaWxlIHNpemVzXG4gKi9cbmNvbnN0IHJlc3BvbnNpdmVHdXR0ZXJzID0gY3NzYFxuICBwYWRkaW5nOiAwO1xuICBAbWVkaWEgKG1pbi13aWR0aDogJHttb2JpbGVCcmVha3BvaW50fXB4KSB7XG4gICAgcGFkZGluZy1sZWZ0OiAke2d1dHRlcldpZHRoIC8gMn1weDtcbiAgICBwYWRkaW5nLXJpZ2h0OiAke2d1dHRlcldpZHRoIC8gMn1weDtcbiAgfVxuYDtcblxuY29uc3QgY29sdW1uU3R5bGVzID0gY3NzYFxuICAke3Jlc3BvbnNpdmVHdXR0ZXJzfTtcbiAgJHtyZXNwb25zaXZlV2lkdGgoY29sdW1uV2lkdGgpfTtcbiAgZmxvYXQ6IGxlZnQ7XG4gIG1pbi1oZWlnaHQ6IDFweDtcbmA7XG5cbmNvbnN0IENvbHVtbiA9ICh7IGNoaWxkcmVuLCBjbGFzc05hbWUgfSkgPT4gKFxuICA8ZGl2IGNzcz17Y29sdW1uU3R5bGVzfSBjbGFzc05hbWU9e2NsYXNzTmFtZX0+XG4gICAge2NoaWxkcmVufVxuICA8L2Rpdj5cbik7XG5cbkNvbHVtbi5kZWZhdWx0UHJvcHMgPSB7XG4gIGNsYXNzTmFtZTogJycsXG59O1xuXG5Db2x1bW4ucHJvcFR5cGVzID0ge1xuICAvKipcbiAgICogQ29sdW1uIGVsZW1lbnRzOiB0aGlzIGlzIHdoZXJlIHRoZSBwYWdlIGNvbnRlbnQgd2lsbCBzaXQsXG4gICAqIHNvIHRoZXNlIGVsZW1lbnRzIGNhbiBiZSBoZWFkaW5ncywgPGRpdj4gZXRjLlxuICAgKi9cbiAgY2hpbGRyZW46IFByb3BUeXBlcy5ub2RlLmlzUmVxdWlyZWQsXG4gIC8qKlxuICAgKiBPcHRpb25hbCBjbGFzcyBuYW1lLlxuICAgKi9cbiAgY2xhc3NOYW1lOiBQcm9wVHlwZXMuc3RyaW5nLFxufTtcblxuZXhwb3J0IGRlZmF1bHQgQ29sdW1uOyJdfQ== */",
    "name": "89bhr6-responsiveGutters-columnStyles",
    "next": undefined,
    "styles": "padding:0;@media (min-width:640px){padding-left:10px;padding-right:10px;}label:responsiveGutters;;width:100%;@media (min-width:640px){width:8.333333333333334%;}/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInJlc3BvbnNpdmVXaWR0aC5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFjeUMiLCJmaWxlIjoicmVzcG9uc2l2ZVdpZHRoLmpzIiwic291cmNlc0NvbnRlbnQiOlsiLyoqIEBqc3gganN4ICovXG5pbXBvcnQgeyBjc3MgfSBmcm9tICdAZW1vdGlvbi9jb3JlJztcblxuaW1wb3J0IGNvbmZpZyBmcm9tICcuL0dyaWRDb25maWcnO1xuXG5jb25zdCB7IG1vYmlsZUJyZWFrcG9pbnQgfSA9IGNvbmZpZztcblxuLyogQHBhcmFtIHtudW1iZXJ9IGNvbHVtbiB3aWR0aCBleHByZXNzZWQgYXMgYSBwZXJjZW50YWdlXG4gKiBUaGUgcmVzcG9uc2l2ZVdpZHRoIENTUyBkZXNjcmliZXMgaG93IGNvbHVtbnMgc2hvdWxkXG4gKiBhbHdheXMgY29sbGFwc2UgdG8gYSBzaW5nbGUtY29sdW1uIGxheW91dCAoMTAwJSB3aWR0aClcbiAqIG9uIG1vYmlsZSBzaXplcy4gQXQgYW55IHNpemUgYWJvdmUgYSBtb2JpbGUgcmFuZ2UsXG4gKiB0aGUgY29sdW1uJ3Mgd2lkdGggd2lsbCBiZSBzZXQgYnkgd2hhdGV2ZXIgbnVtZXJpY1xuICogcGVyY2VudGFnZSB2YWx1ZSBhcmd1bWVudCB0aGF0IGlzIHBhc3NlZCBpbi5cbiAqL1xuY29uc3QgcmVzcG9uc2l2ZVdpZHRoID0gd2lkdGhWYWx1ZSA9PiBjc3NgXG4gIHdpZHRoOiAxMDAlO1xuICBAbWVkaWEgKG1pbi13aWR0aDogJHttb2JpbGVCcmVha3BvaW50fXB4KSB7XG4gICAgd2lkdGg6ICR7d2lkdGhWYWx1ZX0lO1xuICB9XG5gO1xuXG5leHBvcnQgZGVmYXVsdCByZXNwb25zaXZlV2lkdGg7XG4iXX0= */;float:left;min-height:1px;label:columnStyles;/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkNvbHVtbi5qcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFxQndCIiwiZmlsZSI6IkNvbHVtbi5qcyIsInNvdXJjZXNDb250ZW50IjpbIi8qKiBAanN4IGpzeCAqL1xuaW1wb3J0IFByb3BUeXBlcyBmcm9tICdwcm9wLXR5cGVzJztcbmltcG9ydCB7IGNzcywganN4IH0gZnJvbSAnQGVtb3Rpb24vY29yZSc7XG5cbmltcG9ydCBjb25maWcgZnJvbSAnLi9HcmlkQ29uZmlnJztcbmltcG9ydCBjb2x1bW5XaWR0aCBmcm9tICcuL2NvbHVtbldpZHRoJztcbmltcG9ydCByZXNwb25zaXZlV2lkdGggZnJvbSAnLi9yZXNwb25zaXZlV2lkdGgnO1xuXG5jb25zdCB7IGd1dHRlcldpZHRoLCBtb2JpbGVCcmVha3BvaW50IH0gPSBjb25maWc7XG5cbi8qXG4gKiBXZSBvbWl0IGd1dHRlcnMgb24gY29sdW1ucyBieSBkZWZhdWx0IG9uIG1vYmlsZSBzaXplc1xuICovXG5jb25zdCByZXNwb25zaXZlR3V0dGVycyA9IGNzc2BcbiAgcGFkZGluZzogMDtcbiAgQG1lZGlhIChtaW4td2lkdGg6ICR7bW9iaWxlQnJlYWtwb2ludH1weCkge1xuICAgIHBhZGRpbmctbGVmdDogJHtndXR0ZXJXaWR0aCAvIDJ9cHg7XG4gICAgcGFkZGluZy1yaWdodDogJHtndXR0ZXJXaWR0aCAvIDJ9cHg7XG4gIH1cbmA7XG5cbmNvbnN0IGNvbHVtblN0eWxlcyA9IGNzc2BcbiAgJHtyZXNwb25zaXZlR3V0dGVyc307XG4gICR7cmVzcG9uc2l2ZVdpZHRoKGNvbHVtbldpZHRoKX07XG4gIGZsb2F0OiBsZWZ0O1xuICBtaW4taGVpZ2h0OiAxcHg7XG5gO1xuXG5jb25zdCBDb2x1bW4gPSAoeyBjaGlsZHJlbiwgY2xhc3NOYW1lIH0pID0+IChcbiAgPGRpdiBjc3M9e2NvbHVtblN0eWxlc30gY2xhc3NOYW1lPXtjbGFzc05hbWV9PlxuICAgIHtjaGlsZHJlbn1cbiAgPC9kaXY+XG4pO1xuXG5Db2x1bW4uZGVmYXVsdFByb3BzID0ge1xuICBjbGFzc05hbWU6ICcnLFxufTtcblxuQ29sdW1uLnByb3BUeXBlcyA9IHtcbiAgLyoqXG4gICAqIENvbHVtbiBlbGVtZW50czogdGhpcyBpcyB3aGVyZSB0aGUgcGFnZSBjb250ZW50IHdpbGwgc2l0LFxuICAgKiBzbyB0aGVzZSBlbGVtZW50cyBjYW4gYmUgaGVhZGluZ3MsIDxkaXY+IGV0Yy5cbiAgICovXG4gIGNoaWxkcmVuOiBQcm9wVHlwZXMubm9kZS5pc1JlcXVpcmVkLFxuICAvKipcbiAgICogT3B0aW9uYWwgY2xhc3MgbmFtZS5cbiAgICovXG4gIGNsYXNzTmFtZTogUHJvcFR5cGVzLnN0cmluZyxcbn07XG5cbmV4cG9ydCBkZWZhdWx0IENvbHVtbjsiXX0= */",
  },
}
`;

exports[`Grid : Column should render as expected when passed a col(n) method and other styling 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Grid : Column should render as expected when passed a col(n) method and other styling 2`] = `
Object {
  "children": [Function],
}
`;

exports[`Grid : Column should render as expected when passed a col(n) method as a className 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Grid : Column should render as expected when passed a col(n) method as a className 2`] = `
Object {
  "children": [Function],
}
`;
